ospd/ospd.py

```diff
@@ -711,6 +700,32 @@ def get_scan_results_xml(
         logger.debug('Returning %d results', len(results))
         return results
 
+    def get_scan_progress_xml(self, scan_id: str):
```

Comment:
Personally I don't think this method should be in the daemon class but currently i can't find a better place at the moment. XML creation should not be part of the daemon class at all. Maybe just mark it as private for now by changing the name to `_get_scan_progress_xml`